server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -280,14 +282,23 @@ protected static void parseProperties(
                     }
 
                     if (objBuilder.subobjects.value() == false && type.equals(ObjectMapper.CONTENT_TYPE)) {
-                        throw new MapperException(
+                        throw new MapperParsingException(
                             "Object ["
                                 + objBuilder.name()
                                 + "] has subobjects set to false hence it does not support inner object ["
                                 + fieldName
                                 + "]"
                         );
                     }
+                    if (objBuilder.subobjects.value() == false && type.equals(NestedObjectMapper.CONTENT_TYPE)) {
+                        throw new MapperParsingException(
+                            "Object ["
+                                + objBuilder.name()
+                                + "] has subobjects set to false hence it does not support nested object ["
```

Comment:
here you create a separate exception message for nested (vs object) fields, whereas in buildMappers you don't?

Reply:
yea this is odd, it's a consequence of how the check is performed, the two have a different type, but NestedObjectMapper extends ObjectMapper hence one instanceof check covers them both. Another reason is that I found no way to test a possible separate error for nested, because nested type never gets dynamically mapped. If it does it comes from a dynamic template but that does go through parsing and fails earlier during validation (test that I added to DynamicTemplatesTests). All in all I thought it's not wort going the extra mile of adding a separate error message for nested here. I am curious if you have other ideas though.

Reply:
Can we assert then that we never have a NestedObjectMapper there?

Reply:
yea that could work. That should help figuring out how to test that scenario if it ends up being possible :)

Reply:
right, it might hold a surprise :)